client/client.rs

```diff
@@ -183,6 +183,32 @@ impl Client {
         false
     }
 
+    /// Prints a warning and prompt for given action and tasks.
+    /// Returns `Ok(())` if the action was confirmed.
+    fn handle_user_confirmation(&self, action: &str, task_ids: &[usize]) -> Result<()> {
+        // printing warning and prompt
+        println!(
+            "You are trying to {}: {}",
+            action,
+            task_ids
+                .iter()
+                .map(|t| format!("task{}", t.to_string()))
+                .collect::<Vec<String>>()
+                .join(", ")
+        );
+        print!("\nDo you want to continue [Y/n]: ");
+
+        let mut input = String::new();
+        io::stdout().flush().unwrap();
+        io::stdin().read_line(&mut input)?;
+
+        if input.starts_with('n') || input.starts_with('N') {
```

Comment:
This might be a good place to use a `loop`.
It's probably a good idea to check whether the user entered either `Y/y` or `N/n`. Right now it is possible that the user enters any other character than `N/n` and the input will be accepted as if it was an `y`.

Reply:
Should work as intended now, I totally forgot to about the other options :smile: